src/StructuredLogViewer.Core/Timeline/Timeline.cs

```diff
@@ -43,11 +109,254 @@ private void Populate(Build build)
                     Lanes[nodeId] = lane;
                 }
 
-                var block = CreateBlock(node);
-                lane.Add(block);
+                if (includeCpp)
+                {
+                    IEnumerable<Block> blocks = PopulateCppNodes(node);
+                    lane.AddRange(blocks);
+                }
+
+                lane.Add(CreateBlock(node));
             });
         }
 
+        private IEnumerable<Block> PopulateCppNodes(TimedNode node)

```

Comment:
Logically this type of work belongs in an analyzer. After a build is read from a binlog, it is analyzed, and additional (virtual) nodes are added to the tree. See for example:
https://github.com/KirillOsenkov/MSBuildStructuredLog/blob/f1a3384d5e883432b410788a86f454b65499ec25/src/StructuredLogger/Analyzers/BuildAnalyzer.cs#L318

Reply:
I'm OK with leaving this logic here, but if it makes sense, consider moving it to a CppAnalyzer.

Reply:
Created CppAnalyzer to hold all the cpp stuff.